added entry on redirect after logout (#6086)
plone/volto: docs/source/development/redirect-after-logout.md

---
myst:
  html_meta:

>     "description": "How to Redirect to the Login Page After Logout in Volto"

Reviewer: We use sentence case for headings, and minimize wording.

```suggestion
    "description": "Redirect to the login page after logout in Volto"
```

>     "property=og:description": "How to Redirect to the Login Page After Logout in Volto"

Reviewer: ```suggestion
    "property=og:description": "How to redirect to the login page after logout in Volto"
```

>     "property=og:title": "How to Redirect to the Login Page After Logout in Volto"

Reviewer: ```suggestion
    "property=og:title": "Redirect to the login page after logout in Volto"
```

    "keywords": "Volto, Plone, frontend, React, logout, redirect"
---

> # How to Redirect to the Login Page After Logout in Volto

Reviewer: Because the context is Volto, we don't need to repeat it.

```suggestion
# Redirect to the login page after logout
```

> ## Overview 

Reviewer: Avoid empty headings. The following sentence is a good intro.

```suggestion
```

> To redirect users to the login page after they log out in Volto, you need to customize the `PersonalTools` component. Follow these steps:

Reviewer: Plone narrative docs style guide uses one sentence per line.

```suggestion
To redirect users to the login page after they log out in Volto, you need to customize the `PersonalTools` component.
Follow these steps.
```

> ## Steps

Reviewer: It's obvious what comes next.

```suggestion
```

> 

Reviewer: ```suggestion
```

> 1.  **Shadow the component**: Create a new file at `customizations/components/manage/Toolbar/PersonalTools.jsx`, then copy the contents of the original file from {file}`volto/packages/volto/src/components/manage/Toolbar/PersonalTools.jsx` into the new file.

Reviewer 2: I think the best place to do so is in the View, rather than customizing the component as a whole. For example we do redirect to login when unauthorized https://github.com/eea/volto-eea-website-theme/blob/ff514ef4cd7833edcf3956f300092554a0d8ac2e/src/customizations/volto/components/theme/Unauthorized/Unauthorized.jsx#L28

> 1. **Shadow the component**: Create a new file at `customizations/components/manage/Toolbar/PersonalTools.jsx` and copy the contents of the original `PersonalTools.jsx` file from `volto/packages/volto/src/components/manage/Toolbar/PersonalTools.jsx`.

Reviewer: Use MyST markup for files.

```suggestion
1.  **Shadow the component**: Create a new file at `customizations/components/manage/Toolbar/PersonalTools.jsx`, then copy the contents of the original file from {file}`volto/packages/volto/src/components/manage/Toolbar/PersonalTools.jsx` into the new file.
```

> 2. **Modify the logout link**: In the shadowed `PersonalTools.jsx` file, locate the `Link` component that renders the logout button. Modify the `to` prop to include the `return_url` parameter with the desired redirect URL. In this case, we want to redirect to the `/login` page after logout:

Reviewer: ```suggestion
2.  **Modify the logout link**: In the shadowed {file}`PersonalTools.jsx` file, locate the `Link` component that renders the logout button.
    Modify the `to` prop to include the `return_url` parameter with the desired redirect URL.
    In this case, we want to redirect to the `/login` page after logout:
```

```jsx
<Link id="toolbar-logout" to={`${getBaseUrl(pathname)}/logout?return_url=/login`}>
  <Icon
    className="logout"
    name={logoutSVG}
    size="30px"
    title={intl.formatMessage(messages.logout)}
  />
</Link>
```

> 3. **Save the changes**: Save the modified `PersonalTools.jsx` file.

Reviewer: ```suggestion
3.  **Save the changes**: Save the modified {file}`PersonalTools.jsx` file.
```

> 4. **Restart the Volto development server**: You need to restart the Volto development server to see the changes initially, so that it can detect the newly shadowed component.

Reviewer: ```suggestion
4.  **Restart the Volto development server**: You need to restart the Volto development server to load the changes, so that it can detect the newly shadowed component.
```
